frontend/src/containers/PlayerPage/PlayerPage.js

```diff
@@ -65,15 +72,74 @@ class PlayerPage extends React.Component {
       window.location.href = '/library';
     };
 
+    const toggleVideoInformation = () => {
+      // WDB-118
+    };
+
+    const handleVideoDownload = () => {
+      const userId = window.sessionStorage.getItem('id');
+      this.setState({ showDownloadInProgress: true });
+      downloadVideo(video.id, userId)
+        .then((response) => {
+          const contentDisposition = response.headers['content-disposition'];
+          const filename = contentDisposition.split(';')[1].replaceAll('"', '');
+          fileDownload(response.data, filename);
+          this.setState({
+            showDownloadInProgress: false,
+            showDownloadSuccess: true,
+          });
+          console.log(filename);
+        })
+        .catch(() =>
+          this.setState({
+            showDownloadInProgress: false,
+            showDownloadError: true,
+          }),
+        );
+    };
+
     const hidePlaybackError = () => {
-      this.setState({ playbackErrorShowing: false });
+      this.setState({ showPlaybackError: false });
       handleArrowBackClick();
     };
 
+    const renderDownloadSnackbars = () => {
+      if (showDownloadError) {
+        return (
+          <CustomSnackbar
+            message="Ooops.. Something wrong happened. Please try again later"
+            onClose={() => this.setState({ showDownloadError: false })}
+            severity="error"
+            topCenter
+          />
+        );
+      }
+      if (showDownloadInProgress) {
+        return (
+          <CustomSnackbar
+            message="We are pumping the video for you"
```

Comment:
What about "We are downloading the video for you"

Reply:
"We are crunching the video for you"

Reply:
is there a fucking difference? 😄